src/Apps/Order/Components/CheckoutAddress.tsx

```diff
@@ -97,8 +101,49 @@ export const CheckoutAddress: FC<{
     // submit suggested address
   }
 
-  const handleFormSubmit = async (address: Address) => {
-    const { firstLine, secondLine } = formatAddress(address)
+  const handleFormSubmit = async ({ name, ...address }: Address) => {
+    const response = await fetchQuery<any>(
+      relayEnvironment as Environment,
+      graphql`
+        query CheckoutAddressQuery($address: AddressInput!) {
+          verifyAddress(address: $address) {
+            inputAddress {
+              lines
+              address {
+                addressLine1
+                addressLine2
+                city
+                country
+                postalCode
+                region
+              }
+            }
+            suggestedAddresses {
+              lines
+              address {
+                addressLine1
+                addressLine2
+                city
+                country
+                postalCode
+                region
+              }
+            }
+            verificationStatus
+          }
+        }
+      `,
+      {
+        address,
+      },
+      {
+        fetchPolicy: "network-only",
+      }
+    ).toPromise()
+
+    console.log({ response })
+
+    const { firstLine, secondLine } = formatAddress({ name, ...address })
```

Comment:
We use Yup to validate. Here's an example of an async validator: https://github.com/artsy/force/blob/main/src/Components/PhoneNumberInput/validatePhoneNumber.ts

Reply:
See https://github.com/artsy/force/tree/main/src/Components/PhoneNumberInput#validatephonenumber for how to include one in the validation schema

Reply:
Thanks for the resources! For context, there's two types of validation/verification happening in this form: field validation and address verification. We'll be showing inline error messages like "a postal code is required" when field validation fails, but we'll be opening a modal dialog when address verification fails.

Would you recommend that we use Yup to initiate address verification once field validation passes?

<img width="1157" alt="modal dialog" src="https://github.com/artsy/force/assets/44589599/3f942b5f-7f8d-4e26-8552-4d020a1e0b7f">

Reply:
Ah, that looks like it should be it's own fragment container since all this data is tied to UI. But you'll also want to conditionally transition the dialog into this state. You *could* just transition the dialog into that state as a processing step then transition out of it if the query comes back valid or what have you.

Reply:
But yes, as you say: Yup is only used for form validation and this is a distinct kind of validation not tied to fields.

Reply:
I mean it sounds like what should happen is that the verification should be a field on the mutation, you request it as part of the mutation and if needed you transition into this dialog

Reply:
I like the idea of Exchange being responsible for address verification (via Gravity or directly) when the shipping method/address are saved to the order. If verification succeeds, Force won't need to behave any differently than it does today. If verification fails, Force will be notified in the response from the `SetShipment` mutation.

@erikdstock and @fladson I see that you considered this approach in the [tech plan](https://www.notion.so/artsy/Address-Verification-Checkout-flow-51ae42901d094faf8b8c18f6c61de875?pvs=4#e294d7b1847b4602b29c344d97265be4). I'm not proposing we change approaches for this iteration, but for the sake of discussion, I'm curious whether you're still interested in that approach or not.

Reply:
I added a comment on that section [here](https://www.notion.so/Address-Verification-Checkout-flow-51ae42901d094faf8b8c18f6c61de875?d=462d2c67714f4eb7a347be872ea99a8b&pvs=4#87ae84a74051418d838351057a1860d1). I think it's worth revisiting.